### PR TITLE
issue:3518592 Remove reports generated for NDT if NDT file removed

### PR DESCRIPTION
**Pull request by alextabachnik**
Removing report generated for NDT file verification if removed NDT file itself

### CODE REVIEW

File: plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py [hunk 1]
```diff
@@ -381,6 +382,32 @@ def __init__(self):
...
+        :param ndt: name of NDT file
+        '''
+        # find reports related to NDT file name and remove them.
+        # Update reports.json file
+        try:
+            with open(self.reports_list_file, "r", encoding="utf-8") as reports_list_file:
+                # unhandled exception in case reports file was changed manually
+                data = json.load(reports_list_file)
+                data_to_proceed = data.copy()
+                for report_data in data_to_proceed:
```
Comment (shmufeld) [inline plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py:397]: this loop can be outside the "with" statement?

**Review by atolikin94 — APPROVED** [review]
Don't really understand why do we need to remove reports when NDT was remover, but the code looks good.

File: plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py (hunk 1 above)
Comment (alextabachnik) [reply plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py:397]: Possible to open file for read and  read data and then to close. 
With doing the same. In this case both options are valid.

Comment (shmufeld) [reply plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py:397]: you right. it's possible but I thought it will be cleaner...never mind.

**Review by shmufeld — APPROVED** [review]

**Review by kedeme — APPROVED** [review]

**Review by kedeme — APPROVED** [review]

**Pull request merged**